File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/AudioRoom/View/ApplyUserList/PanoMyApplyView.m
//
//  PanoMyApplyView.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/19.
//

#import "PanoMyApplyView.h"
#import "PanoDefine.h"

@implementation PanoMyApplyView
{
    UILabel *nameLabel;

    UIButton *cancelBtn;
}

- (void)initViews {
    
    self.layer.cornerRadius = 5;
    self.layer.masksToBounds = true;
    
    self.backgroundColor = [UIColor whiteColor];
    nameLabel = [UILabel new];
    nameLabel.font = fontMedium();
    nameLabel.textColor = defaultTextColor();
    nameLabel.text = NSLocalizedString(@"已申请上麦，等候通过···",  nil);
    [nameLabel sizeToFit];
    [self addSubview:nameLabel];
    
    cancelBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    [cancelBtn setTitle:NSLocalizedString(@"取消", nil) forState: UIControlStateNormal];
    
    [cancelBtn setTitleColor:[UIColor pano_colorWithHexString:@"#69B4F9"] forState:UIControlStateNormal];
    [cancelBtn setTitleColor:[UIColor pano_colorWithHexString:@"#69B4F9"] forState:UIControlStateHighlighted];
    [cancelBtn addTarget:self action:@selector(cancelApplyChat) forControlEvents:UIControlEventTouchUpInside];
    [cancelBtn sizeToFit];
    [self addSubview:cancelBtn];
}

- (void)initConstraints {
    [nameLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        
        make.left.bottom.mas_equalTo(0).insets(UIEdgeInsetsMake(0, 8, 8, 0));
        make.right.equalTo(cancelBtn.mas_left).offset(-defaultMargin);
    }];
    [cancelBtn mas_makeConstraints:^(MASConstraintMaker *make) {
        make.right.mas_equalTo(0).offset(-defaultMargin);
        make.centerY.equalTo(nameLabel);
    }];
}

- (void)showInView:(UIView*)view {
    [view addSubview:self];
    CGFloat viewHeight = (isIphoneX() ? 44 : 20) + 40;
    [self mas_remakeConstraints:^(MASConstraintMaker *make) {
        make.centerX.mas_equalTo(0);
        make.bottom.equalTo(view.mas_top);
        make.height.mas_equalTo(viewHeight);
    }];
    [UIView animateWithDuration:0.3 animations:^{
        self.transform = CGAffineTransformMakeTranslation(0, viewHeight);
    }];

}

- (void)dismiss {
    [UIView animateWithDuration:0.3 animations:^{
        self.transform = CGAffineTransformIdentity;
    } completion:^(BOOL finished) {
        [self removeFromSuperview];
    }];
}

- (void)cancelApplyChat {
    cancelBlock();
}


@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/AudioRoom/View/ApplyUserList/PanoMyApplyView.swift
//
//  PanoMyApplyView.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/11/3.
//

import UIKit

class PanoMyApplyView: PanoBaseView {

    var nameLabel : UILabel!
    
    var cancelBtn : UIButton!
    
    var cancelBlock: (() -> Void)!
    
    override func initViews() {
        
        self.layer.cornerRadius = 5;
        self.layer.masksToBounds = true;
        
        self.backgroundColor = UIColor.white
        nameLabel = UILabel()
        nameLabel.font = FontMedium
        nameLabel.textColor = DefaultTextColor
        nameLabel.text = NSLocalizedString("已申请上麦，等候通过···", comment: "")
        nameLabel.sizeToFit()
        self.addSubview(nameLabel)
        
        cancelBtn = UIButton()
        cancelBtn.setTitle(NSLocalizedString("取消", comment: ""), for: .normal)
        cancelBtn.setTitleColor(UIColor("#69B4F9"), for: .normal)
        cancelBtn.setTitleColor(UIColor("#69B4F9"), for: .highlighted)
        cancelBtn.addTarget(self, action: #selector(cancelApplyChat), for: .touchUpInside)
        cancelBtn.sizeToFit()
        self.addSubview(cancelBtn)
    }
    
    override func initConstraints() {
        nameLabel.snp.makeConstraints { (make) in
            make.left.bottom.equalToSuperview().inset(UIEdgeInsets(top: 0, left: 8, bottom: 8, right: 0))
            make.right.equalTo(cancelBtn.snp.left).offset(-defaultMargin)
        }
        cancelBtn.snp.makeConstraints { (make) in
            make.right.equalToSuperview().offset(-defaultMargin)
            make.centerY.equalTo(nameLabel)
        }
    }
    
    func showInView(view: UIView){
        view.addSubview(self)
        let viewHeight = (PanoIsIphoneX() ? 44 : 20) + 40
        self.snp.remakeConstraints { (make) in
            make.centerX.equalToSuperview()
            make.bottom.equalTo(view.snp.top)
            make.height.equalTo(viewHeight)
        }
        UIView.animate(withDuration: 0.3) {
            self.transform = CGAffineTransform(translationX: 0, y: CGFloat(viewHeight))
        }
    }
    
    func dismiss() {
        UIView.animate(withDuration: 0.3) {
            self.transform = CGAffineTransform.identity
        } completion: { (flag) in
            self.removeFromSuperview()
        }
    }
    
    @objc func cancelApplyChat() {
        cancelBlock();
    }
}
